src/main/java/com/alvarium/DefaultSdk.java

```diff
@@ -1,14 +1,27 @@
 package com.alvarium;
 
+import com.alvarium.annotators.Annotator;
+import com.alvarium.streams.StreamException;
+import com.alvarium.streams.StreamProvider;
+import com.alvarium.streams.StreamProviderFactory;
 import com.alvarium.utils.PropertyBag;
 
-public class DefaultSdk implements Sdk {
-
-  DefaultSdk() {
-    this.init();
-  }
+import org.apache.logging.log4j.Logger;
 
-  private void init() {
+public class DefaultSdk implements Sdk {
+  private final Annotator[] annotators;
+  private final SdkInfo config;
+  private final StreamProvider stream;
+
+  DefaultSdk(Annotator[] annotators, SdkInfo config, Logger logger) throws StreamException {
+    this.annotators = annotators;
+    this.config = config;
+
+    // init stream
+    final StreamProviderFactory streamFactory = new StreamProviderFactory();
+    this.stream = streamFactory.getProvider(this.config.getStream());
+    this.stream.connect();
+    logger.debug("stream provider connected successfully.");
```

Comment:
You're going to want to set `logger` to a private member for use in the other methods. For example.

https://github.com/project-alvarium/alvarium-sdk-go/blob/ab0e28e29948aa6653018af993663175775c1c7a/pkg/sdk.go#L90

I'm not going to block the PR on this, just calling it out.